### ISSUE TITLE
object-literal-sort-keys natural sort order

### ISSUE THREAD

**Issue by fibo** [palantir/tslint#3977]
### Bug Report

- __TSLint version__: 5.10.0
- __TypeScript version__: 2.9.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
     'is-6': is6,
      'is-7': is7,
      'is-8': is8,
      'is-9': is9,
      'is-10': is10,
      'is-11': is11,
```

with `tslint.json` configuration:

```json
{
  "extends": ["tslint:latest", "tslint-react"],
  "rules": {
    "interface-name": false,
    "quotemark": [true, "single", "jsx-double"],
    "semicolon": [true, "never"],
    "trailing-comma": false,
  // override tslint-react rules here
    "jsx-wrap-multiline": false
  }
}
```

#### Actual behavior

Sorting does not consider numbers.

#### Expected behavior

In my opinion `is-9` should be before `is-10`.

Not a big issue, I just ignored it with comments, 

![screen shot 2018-06-20 at 15 18 04](https://user-images.githubusercontent.com/176730/41660759-40a5a276-749d-11e8-91e0-466031ab69ff.png)

**Comment by NaridaL** [comment]
The issue is that it's currently sorting alphabetically, while you want a "natural sort". String.localeCompare actually optionally supports that, so it should be easy to add an option to this rule to support that.

**Comment by fibo** [comment]
In my opinion it should be the default, it is more DWIM. But, I understand that it could be annoying for people who will need to apply few modifications in their props order.

@NaridaL thank you very much I saw in that thread an answer with a natural sort implementation that will help a lot.

I think I could do a PR, the question is: **should natural sort be optional or default**?

Ahaha, I had a look to the rule code, it is a really neat code base, but it was really funny to find out that *objectLiteralSortKeysRule.ts* itself [tslint:disable:object-literal-sort-keys comment](https://github.com/palantir/tslint/blob/5.10.0/src/rules/objectLiteralSortKeysRule.ts#L34)

**Comment by giladgray** [comment]
@fibo hey if you want to go ahead and add support for natural sort via an option then please go ahead. if you _don't_ want to contribute this then please close this issue.

alphabetical sorting is trivial but natural sort is much more expensive so i'm inclined to do nothing here. disable the sorting as necessary to achieve the ordering you desire. 

that's what we do in the rule code itself: disable sorting so we can order the properties in a more logical sense that tells a better story about the rule.

**Comment by fibo** [comment]
@giladgray I would like to do a PR, but I also think it is difficult I could find the time to do it in the near future. I wrote an issue about this in my repo, if I will do a PR I will notify you.

By now I close this issue, to avoid noise.
